digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/lane/LaneDao.scala

```diff
@@ -0,0 +1,446 @@
+          vvhTimeStamp = row.vvhTimeStamp, geomModifiedDate = row.geomModifiedDate, attributes = attributeValues)
+
+    }.values.toSeq
+  }
+
+
+  def laneRowToProperty( laneRows: Iterable[LaneRow]): Seq[LaneProperty] = {
+
+    val laneCodeAttribute = Seq(LaneProperty( "lane_code", Seq(LanePropertyValue(laneRows.head.laneCode))))
+
+    val props = laneRows.groupBy(_.value.publicId).map { case (key, rows) =>
+      val row = rows.head
+
+      LaneProperty(
+        publicId = row.value.publicId,
+
+        values = rows.flatMap(laneRow =>
+          laneRow.value.propertyValue match {
+            case Some(value) => Some(LanePropertyValue(value))
+            case _ => None
+          }
+        ).toSeq
+      )
+
+    }.toSeq
+
+    props ++ laneCodeAttribute
+  }
+
+  /**
+    * Updates validity of lane in db.
+    */
+  def updateLaneExpiration(id: Long, username: String): Unit = {
+
+    sqlu"""UPDATE LANE
+          SET valid_to = sysdate,  expired_date = sysdate, expired_by = $username
+          WHERE id = $id
+    """.execute
+
+  }
+
+
+  def updateLane( id: Long, lane: PieceWiseLane, username: String) = {
+    val laneCode =lane.laneAttributes.properties.find( _.publicId == "lane_code" ).head.values
+
+    sqlu"""UPDATE lane
+          SET lane_code =${laneCode.head.value.toString},  modified_date = sysdate, modified_by = $username
+          WHERE id = $id
+          AND lane_code = ${lane.linkId}
+    """.execute
+  }
+
+
+  def updateLanePosition (newIncomeLane: PersistedLane): Unit ={
+    sqlu"""UPDATE LANE_POSITION
+          SET start_measure = ${newIncomeLane.startMeasure}, end_measure = ${newIncomeLane.endMeasure},
+              modified_date = sysdate, adjusted_timestamp = ${newIncomeLane.vvhTimeStamp}
+          WHERE link_id = ${newIncomeLane.linkId}
+    """.execute
+  }
+
+
+  def createLanePosition(persistedLane: PersistedLane, username: String): Long = {
+    val lanePositionId = Sequences.nextPrimaryKeySeqValue
+
+    sqlu"""INSERT INTO LANE_POSITION (id, side_code, start_measure, end_measure, link_id, adjusted_timestamp)
+          VALUES ( $lanePositionId, ${persistedLane.sideCode}, ${persistedLane.startMeasure}, ${persistedLane.endMeasure},
+              ${persistedLane.linkId}, ${persistedLane.vvhTimeStamp})
+    """.execute
+
+    lanePositionId
+  }
+
+  def createLanePositionRelation (laneId: Long, lanePositionId: Long): Unit = {
+    sqlu"""INSERT INTO LANE_LINK (lane_id, lane_position_id)
+          VALUES ($laneId, $lanePositionId )
+    """.execute
+  }
+
+  def createLane ( newIncomeLane: PersistedLane, username: String): Long = {
+
+    val laneId = Sequences.nextPrimaryKeySeqValue
+
+    sqlu"""INSERT INTO LANE (id, lane_code, created_date, created_by, municipality_code)
+          VALUES ($laneId, ${newIncomeLane.laneCode}, sysdate, ${username}, ${newIncomeLane.municipalityCode} )
+    """.execute
+
+    laneId
+  }
```

Comment:
Let's put the create similar to the already existent code and delete createLanePositionRelation and createLanePosition

```scala
  def createLaneV2(newIncomeLane: PersistedLane, username: String): Long = {
    val laneId = Sequences.nextPrimaryKeySeqValue
    val lanePositionId = Sequences.nextPrimaryKeySeqValue

    sqlu"""
        insert all
          into LANE (id, lane_code, created_date, created_by, municipality_code)
          values ($laneId, ${newIncomeLane.laneCode}, sysdate, ${username}, ${newIncomeLane.municipalityCode} )

          into LANE_POSITION (id, side_code, start_measure, end_measure, link_id, adjusted_timestamp)
          values ( $lanePositionId, ${newIncomeLane.sideCode}, ${newIncomeLane.startMeasure}, ${newIncomeLane.endMeasure},
                   ${newIncomeLane.linkId}, ${newIncomeLane.vvhTimeStamp})

          into LANE_LINK (lane_id, lane_position_id)
          values ($laneId, $lanePositionId )
        select * from dual
      """.execute

    laneId
  }

Reply:
Changeed as requested